src/gpu/sycl/ref_shuffle.cpp

```diff
@@ -0,0 +1,147 @@
+    conf_.axis_size = axis_size();
+    conf_.ndims = ndims();
+    auto dim = desc()->dst_desc.dims;
+    conf_.ndims_d = desc()->dst_desc.ndims;
+    conf_.outer_size = utils::array_product(dim, conf_.axis);
+    conf_.inner_size = utils::array_product(dim + conf_.axis + 1, conf_.ndims_d - conf_.axis - 1);
+    conf_.group_size = group_size();   
+    conf_.wg_size = 32;
+    conf_.MB = MB();
+    conf_.C = C();
+    
+    const bool has_spatial = utils::one_of(data_d.ndims(), 3, 4, 5);
+    if (has_spatial) {
+       conf_.H = H();
+       conf_.W = W();
+       conf_.D = D();
+       conf_.HW = conf_.H * conf_.W;
+       conf_.SP = conf_.D * conf_.HW;
+    }
+    conf_.stat_md=sycl_md_t(src_md(0)); 
+    conf_.work_amount = memory_desc_wrapper(src_md()).nelems();
+    conf_.src_md = sycl_md_t(src_md(0));
+    conf_.dst_md = sycl_md_t(dst_md(0));
+    conf_.tag = dat_tag_;
+
+    conf_.stride_m = data_d.blocking_desc().strides[0];
+    conf_.block_size = data_d.blocking_desc().strides[conf_.ndims - 1];
+
+    conf_.transpose_row = is_fwd() ? conf_.group_size
+                                          : conf_.axis_size / conf_.group_size;
+    conf_.transpose_col = is_fwd()
+             ? conf_.axis_size / conf_.group_size
+             : conf_.group_size;
+
+    int d2 = (conf_.block_size > conf_.C) ? 1 : conf_.block_size;
+
+    const int t_work
+            = (conf_.MB) * ((conf_.C / d2)) * conf_.SP;
+    const int wg_work = conf_.wg_size * conf_.block_size;
+    const int wg_cnt = (t_work + wg_work - 1) / wg_work;
+    conf_.nthr = wg_cnt * conf_.wg_size;
+ 
+    if (memory_desc_wrapper(src_md()).nelems() % conf_.block_size != 0)
+         return status::unimplemented;
+
+    conf_.prop_kind = desc()->prop_kind;
+
+     return status::success;
+}
+
+status_t ref_shuffle_t::init(engine_t *engine) {
+
+    if (pd()->conf_.axis == 1 && one_of(pd()->conf_.tag, nchw, ncdhw)) {
+        const auto kid = ::sycl::get_kernel_id<shuffle_kernel_vec1_t>();
+        CHECK(create_kernel(engine, kid, &kernel_));
+    } else if (pd()->conf_.axis == 1 && one_of(pd()->conf_.tag, nhwc, ndhwc)) {
+        const auto kid = ::sycl::get_kernel_id<shuffle_kernel_vec2_t>();
+        CHECK(create_kernel(engine, kid, &kernel_));
+    } else {
+        const auto kid = ::sycl::get_kernel_id<shuffle_kernel_vec3_t>();
+        CHECK(create_kernel(engine, kid, &kernel_));
+    }
+    return status::success;
+}
+
+
+status_t ref_shuffle_t::execute(const exec_ctx_t &ctx)const{
+parallel_for(ctx, kernel_, [&](::sycl::handler &cgh) {
+    auto src_mem_arg = pd()->is_fwd() ? CTX_IN_SYCL_KERNEL_MEMORY(DNNL_ARG_SRC)
+                                      : CTX_IN_SYCL_KERNEL_MEMORY(DNNL_ARG_DIFF_DST);
+    
+    auto dst_mem_arg =  pd()->is_fwd() 
+             ? CTX_OUT_SYCL_KERNEL_MEMORY(DNNL_ARG_DST) 
+             : CTX_OUT_SYCL_KERNEL_MEMORY(DNNL_ARG_DIFF_SRC); 
+    int axis = pd()->axis();
+    auto nelems_A = memory_desc_wrapper(pd()->src_md()).nelems();
```

Comment:
There are a lot of unused variable warnings, please address them

Reply:
Removed